public/js/model/check-record-changes.js

```diff
@@ -0,0 +1,800 @@
+
+    });
+
+    $('form').submit(function () {
+
+        var form = this,
+            formValidator = $(form).data('bootstrapValidator'),
+            formData = $(form).find(":input").serializeArray(),
+            formUrl = $(this).attr('action').replace('save','change');
+
+
+        if (resolvedVersionNo) {
+            formUrl += '/' + resolvedVersionNo;
+        }
+
+        $.ajax({
+            method: 'post',
+            url: formUrl,
+            dataType: 'json',
+            data: formData
+        })
+        .done(function(data, textStatus, jqXHR) {
+
+            if (!data.hasChanged) {
+
+                if (formValidator.isValid()) {
+
+                    // since there are no change, submit the form, proceed with normal save operation
+                    return form.submit();
+
+                } else {
+
+                    return formValidator.validate();
+
+                }
+
+            }
+
+            // since there is a change, lets load a modal to highlight the changes and provide user with options for the next step
+            $('#linzModal').modal().load('/admin/merge-data-conflict-guide', function () {});
+
+            Object.keys(data.diff).forEach(function (fieldName) {
+
+                var formField = $(form).find(':input[name="' + fieldName + '"]');
+
+                if (!formField.length) {
+                    // exit if field is not found!
+                    return;
+                }
+
+                linz[formField.attr('data-linz-conflict-handler')](fieldName, data.diff[fieldName], formField, data, form, formValidator);
+
+            });
+
+        })
+        .fail(function(jqXHR, textStatus, errorThrown ) {
+
+            alert('An error has occured while attempting to check if this record has been editted by other user.');
+            return false;
+
+        });
+
+        return false;
+    });
+
+    function standardInputFieldConflictHandler(fieldName, fieldType, formField, data, form, formValidator) {
+
+        var yourChangeLabel = data.yourChange[fieldName],
+            theirChangeLabel = data.theirChange[fieldName];
+
+        if (fieldType === 'date' || fieldType === 'datetime') {
+
+            if (yourChangeLabel !== '') {
+                try {
+                    yourChangeLabel = moment(new Date(yourChangeLabel).toISOString()).format('DD/MM/YYYY');
+                } catch (e) {}
+            }
+
+            if (theirChangeLabel !== '') {
+                try {
+                    theirChangeLabel = moment(new Date(theirChangeLabel).toISOString()).format('DD/MM/YYYY');
+                } catch (e) {}
+            }
+
+        }
+
+        yourChangeLabel = formatTextFieldLabel(yourChangeLabel);
+        theirChangeLabel = formatTextFieldLabel(theirChangeLabel);
+
+        if (!formField.parent().hasClass('has-change')) {
+
+            formField.wrap('<div class="input-group has-conflict has-change"></div>');
```

Comment:
@sandytrinh, do you think it's worth using mustache or something similar for this? Put the HTML into `<script type="template">` tags and rendering them first?